lib/inspec/resources/users.rb

```diff
@@ -425,10 +463,26 @@ def credentials(username)
         multiple_values: false
       ).params
 
+      cmd = inspec.command("echo $((($(date +%s) - $(date -d '#{params["Last password change"]}' '+%s'))/86400))")
+      if cmd.exit_status != 0
+        dayslastset = nil
+      else
+        dayslastset = convert_to_i(cmd.stdout.chomp)
+      end
+
+      cmd = inspec.command("lastb -w -a | awk '{print $1}' | grep #{username} | wc -l")
+      if cmd.exit_status != 0
+        badpasswordattempts = nil
+      else
+        badpasswordattempts = convert_to_i(cmd.stdout.chomp)
+      end
```

Comment:
```ruby
# lift assignment
badpasswordattempts = if cmd.exit_status != 0
                        nil
                      else
                        convert_to_i(cmd.stdout.chomp)
                      end

# inline conditional
badpasswordattempts = cmd.exit_status != 0 ? nil :convert_to_i(cmd.stdout.chomp)

# simplify
badpasswordattempts = convert_to_i(cmd.stdout.chomp) if cmd.exit_status == 0
```

Reply:
Thanks for this ruby expert tips :) 
i will do the requested updates.